docs/CODE_GUIDELINES.md

```diff
@@ -70,7 +70,7 @@ Conventions can be bent or broken in the interest of making code more readable a
 
 ## 2. Language standard
 
-We currently target the C++14 language standard. Do use C++14 features when possible. Do not use C++17 features.
+We currently target the C++17 language standard. Do use C++17 features when possible (and supported by all target platforms). Do not use C++20 features.
```

Comment:
Do we know of any c++17 features that aren't supported by all platforms?

Reply:
I can't recall which feature it was but I faced a Darwin build issue a while ago in Jenkins (Google pointed at not fully implemented in macOS). Might try to track that PR later if I remember what I was touching back then...

Anyway, I don't see the added comment as a big deal. Is it?

Reply:
The best we can run off would be something like https://en.cppreference.com/w/cpp/17 i guess.

Reply:
Thanks, now I remember. It was the usage of the execution header (parallel execution) which failed to compile in clang (ref https://stackoverflow.com/questions/60859395/unable-to-compile-c17-with-clang-on-mac-osx)